### PR TITLE
Some cleanup and testing

### PR DESCRIPTION
**Pull request by clonker**

### CODE REVIEW

File: sktime/covariance/covariance.py [hunk 1]
```diff
@@ -401,16 +402,18 @@ def fit(self, data, lagtime=None, weights=None, n_splits=None, column_selection=
...
+                wsplit = timeshifted_split(weights, lagtime=lagtime, n_splits=n_splits)
 
         if self.is_lagged:
-            for (x, y), w in zip(timeshifted_split(data, lagtime=lagtime, n_splits=n_splits), wsplit):
+            t = 0
+            for (x, y), (w, _) in zip(timeshifted_split(data, lagtime=lagtime, n_splits=n_splits), wsplit):
                 if lazy_weights:
                     w = weights.weights(x)
-                # weights can weights be shorter than actual data
+                # weights can weights be longer than actual data
```
Comment (brookehus) [inline sktime/covariance/covariance.py:412]: "weights can be..." ?

File: sktime/decomposition/tica.py [hunk 2]
```diff
@@ -212,5 +212,7 @@ def fit_from_covariances(self, covariances: Union[Covariance, CovarianceModel]):
         if not covariances.symmetrized:
             raise ValueError("The covariance model must be estimated such that the "
                              "autocorrelations are symmetric!")
+        if covariances.cov_0t is None:
+            raise ValueError("The covariance model must contain C0t!")
```
Comment (brookehus) [inline sktime/decomposition/tica.py:216]: !

File: sktime/covariance/covariance.py (hunk 1 above)
Comment (clonker) [reply sktime/covariance/covariance.py:412]: yes! thanks!

File: sktime/decomposition/tica.py (hunk 2 above)
Comment (clonker) [reply sktime/decomposition/tica.py:216]: It is important!

**Pull request merged**